AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -991,6 +1026,90 @@ - (CGRect)convertRect:(CGRect)rect toNode:(ASDisplayNode *)node
   return CGRectApplyAffineTransform(rect, flattenedTransform);
 }
 
+#pragma mark - Layout Transition
+
+- (void)transitionLayoutWithAnimation:(BOOL)animated
+{
+  [self transitionLayoutThatFits:_constrainedSize animated:animated];
+}
+
+- (void)transitionLayoutThatFits:(ASSizeRange)constrainedSize animated:(BOOL)animated
+{
+  [self invalidateCalculatedLayout];
+  _deferImmediateHierarchyManagement = YES;
```

Comment:
I am not certain, but does it make sense to factor the hierarchy management into a second method call, and call it only where we do not flip this flag on?